### PR TITLE
Major auto-evo framework updates for species splitting

### LINKED ISSUE #915: Auto-Evo for player
I don't know if this is intended, but Auto-Evo doesn't work for the player. The player's individual successes should have an impact, but their choices in species creation should too, because not all members of the player's species are accounted for in the current system - just the player him/herself.

If needed, the minimum population to thrive could be increased to expand play-time. Auto-Evo is REALLY beneficial to a species population (just load up Free-Build and look at those numbers!) and should thus be given to the player to avoid an in-balance between them and the AI.

To be honest, this may not be the best place to post this thought, but either way, thanks for reading!

### LINKED ISSUE #2096: Add species splitting to auto-evo system
Right now the only species "splitting" that happens is when the player enters the editor a mutated copy of them is added to the patch they are in. We should get a proper species splitting algorithm done sooner than later as some other stuff depend on this.

A bunch of has been said about how this should work: https://forum.revolutionarygamesstudio.com/t/population-system-species-splitting/494

But I think to get something done a basic algorithm should be added and we can iterate on it later:
- Doesn't affect the player species
- Basic logic is: when trying out mutations detect if there are 2 good mutations that are good for different patches of a species, if so, and population is high enough, split the species into two with the original one picking up the better mutation and a new species being generated with the slightly worse mutation. The population should be split among so that which ever mutation does better in a patch is the species that gets the population from that patch.

### PR DESCRIPTION
**Pull request by hhyyrylainen**
**Brief Description of What This PR Does**

Added species splitting, new species splitting when low number, and auto-evo for player
also disabled the mutated copy of player appearing each time

Has a GUI to show auto-evo (upcoming) results to the player to allow making informed decisions. May need to tweak the reproduction population bonus in the future. And also added a tutorial on the second time the player goes to the editor to highlight that they should look at it.

It seems that cloned mutated species where marked as PlayerSpecies, which might indicate that there's a serious bug in master causing species to not actually mutate or spread. Fixed in this PR.

**Related Issues**

closes #2096
closes #915

**Progress Checklist**

Note: before starting this checklist the PR should be marked as non-draft.

- [x] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [x] Initial code review passed (this and further items should not be checked by the PR author)
- [ ] Functionality is confirmed working by another person (see above checklist link)
- [ ] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author.

### CODE REVIEW

**Review by Kasterisk — CHANGES_REQUESTED** [review]

File: src/microbe_stage/editor/MicrobeEditor.tscn [hunk 1]
```diff
@@ -2176,6 +2362,7 @@ HideCancelButton = true
 DialogText = "DISCONNECTED_ORGANELLES_TEXT"
 
 [node name="Report" type="Control" parent="MicrobeEditorGUI"]
+visible = false
```
[the remark this answers is not in the document]
Comment (hhyyrylainen) [reply src/microbe_stage/editor/MicrobeEditor.tscn:2365]: good catch, I tried to be careful with putting these back.

File: src/auto-evo/steps/IncreaseBiodiversity.cs [hunk 2]
```diff
@@ -0,0 +1,159 @@
+﻿namespace AutoEvo
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    /// <summary>
+    ///   Attempts to increase the biodiversity of a patch by force-splitting an existing species there or creating a
+    ///   new offshoot species from a
```
[the remark this answers is not in the document]
Comment (hhyyrylainen) [reply src/auto-evo/steps/IncreaseBiodiversity.cs:9]: added the missing few last words

**Review by Kasterisk — APPROVED** [review]
Maybe @adQuid can review more in-depth about the auto-evo but to me this looks good. Though I really think we need to find a better place or at least a collapse feature for the auto evo prediction panel, maybe could be put in the statistics menu once that's implemented to alleviate GUI crowding.

**Comment by hhyyrylainen** [comment]
Yeah, some GUI rework would be nice. I didn't put it in the organism statistics panel as the compound balances can take up all the remaining space so I'd need to reshuffle things there and make sure scrollbars work fine in there (also the tutorial would need special logic to make sure the auto-evo results are scrolled into view automatically when the tutorial activates).

**Comment by Kasterisk** [comment]
Ah I mean the general statistics menu that will be accessible through the main menu and bottom left buttons in the editor which is currently disabled. I imagine it would be quite a big feature to implement in the future.

**Comment by hhyyrylainen** [comment]
At least part of the auto-evo prediction needs to be always visible when editing, otherwise the player might forget and mess over their species (though I'm about to open an issue about rebalancing the auto-evo algorithm now that the player is affected by it after this is merged).

**Comment by hhyyrylainen** [comment]
I'll merge this now as no changes required reviews have come in and this is important to get in early to allow time for tweaking.

**Pull request merged**